src/Helper/HeadMeta.php

```diff
@@ -474,7 +473,11 @@ public function setCharset($charset)
         $item->charset = $charset;
         $item->content = null;
         $item->modifiers = [];
-        $this->isValid($item);
+
+        if (! $this->isValid($item)) {
+            throw new Exception\RuntimeException('XHTML* doctype has no attribute charset; please use appendHttpEquiv()');
```

Comment:
You don't like the `InvalidArgumentException`?! 😉

Reply:
updated to InvalidArgumentException